modules/features/manatime_feature_timesheets/plugins/relationships/timesheet_node.inc

```diff
@@ -14,33 +14,77 @@ $plugin = array(
   'keyword' => 'node',
   'context' => 'timesheet_node_from_user_context',
   'edit form' => 'timesheet_node_from_user_settings_form',
-  'required context' => new ctools_context_required(t('User'), 'entity:user'),
+  'required context' => array(
+    new ctools_context_required(t('User'), 'entity:user'),
+    new ctools_context_required(t('String'), 'string'),
+  ),
 );
 
 /**
  * Return a new context based on an existing context.
  */
 function timesheet_node_from_user_context($context, $conf) {
-  if (empty($context->data)) {
+  if (empty($context[0]->data)) {
     return ctools_context_create_empty('entity:node', NULL);
   }
-  if (date('w') == 1) {
-    $beginning = new DateTime('now');
+
+  $uid = $context[0]->data->uid;
+  // If the url does not have dates, get the current week's timesheet.
+  if (empty($context[1]->data)) {
+    $beginning = date('w') == 1 ? new DateTime('now') : new DateTime('last monday');
+    $end = $beginning->getTimestamp() + 576000;
+    $dates = array($beginning->format('Y-m-d'), gmdate('Y-m-d', $end));
+    return get_timesheet(0, $dates, $dates, 'BETWEEN', $uid);
   }
+
   else {
-    $beginning = new DateTime('last monday');
+    // Get the timesheets with dates given.
+    $dates = explode("-", $context[1]->data);
+    if (validate_date_format($dates[0]) && validate_date_format($dates[1])) {
+      $beginning = (new DateTime($dates[0]))->format('Y-m-d');
```

Comment:
http://php.net/manual/es/datetime.construct.php returns FALSE (or throws and exception) in case of error. Please try to use that instead of a custom validate function. If FALSE, you can catch it in if; if exception, use try-catch